src/odemis/acq/align/test/fastem_test.py

```diff
@@ -0,0 +1,185 @@
+        cls.stage = model.getComponent(
+            role="stage")  # TODO replace with stage-scan when ROA conversion method available
+        cls.ccd = model.getComponent(role="diagnostic-ccd")
+        cls.focuser = model.getComponent(role="diagnostic-cam-focus")
+        cls.beamshift = model.getComponent(role="ebeam-shift")
+        cls.det_rotator = model.getComponent(role="det-rotator")
+
+    def setUp(self):
+
+        self.good_focus = -70e-6  # position where the image of the multiprobe is displayed in focus [m]
+        self.good_mp_position = (650, 450)  # position for multiprobe where mp is aligned with mppc detector [px_dc]
+        self.ccd.updateMetadata(
+            {model.MD_FAV_POS_ACTIVE:
+                {"x": self.good_mp_position[0],
+                 "y": self.good_mp_position[1],
+                 "z": self.good_focus}}
+                                )
+
+        # move the stage so that the image is in focus
+        self.focuser.moveAbs({"z": self.good_focus}).result()
+
+    def test_optical_autofocus(self):
+        """Run the optical autofocus calibration. Can also be tested with simulator."""
+
+        calibrations = [OPTICAL_AUTOFOCUS]
+
+        # move the stage so that the image is out of focus
+        self.center_position = -30e-6
+        self.focuser.moveAbs({"z": self.center_position}).result()
+
+        # Run auto focus
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        # foc_pos, foc_lev = f.result(timeout=900)  # TODO make use of the focus level?
+        e = f.result(timeout=900)
+
+        self.assertIsNone(e)  # check no exceptions were returned
+        # check that z stage position is close to good position
+        # Note: This accuracy is dependent on the value chosen for the magnification on the lens.
+        numpy.testing.assert_allclose(self.focuser.position.value["z"], self.good_focus, atol=2e-6)
+
+    def test_image_translation_prealign(self):
+        """Run the image translation prealing calibration. Can also be tested with simulator.
+        It calibrates the descanner offset."""
+
+        calibrations = [IMAGE_TRANSLATION_PREALIGN]
+
+        # get current descanner offset
+        descan_offset_cur = self.descanner.scanOffset.value
+
+        # Run image translation pre-align
+        f = fastem.align(self.scanner, self.multibeam, self.descanner, self.mppc, self.stage, self.ccd,
+                         self.beamshift, self.det_rotator, calibrations)
+
+        e = f.result(timeout=900)
+
+        self.assertIsNone(e)  # check no exceptions were returned
+
+        # get the calibrated descanner offset
+        descan_offset_calib = self.descanner.scanOffset.value
+
+        # check the calibrated descan offset is different from the previous offset
+        self.assertNotEqual(descan_offset_cur, descan_offset_calib)
```

Comment:
If you run it on hardware the descanner offset should actually change, so will this test case then fail?

Reply:
aha, I didn't read the "not" I read it as `self.assertEqual`.